### ISSUE TITLE
x/pkgsite: The new index is hard to read

### ISSUE THREAD

**Issue by brackendawson** [golang/go#41287]
### What is the URL of the page with the issue?
https://pkg.go.dev/net/http?tab=doc

### What is your user agent?
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36

### Screenshot
![image](https://user-images.githubusercontent.com/1909753/92595822-daece680-f29c-11ea-8371-ab72c9933a2a.png)

### What did you do?
I was looking for the name of a function that does what I want to do.
 
### What did you expect to see?
Expected to quickly find the function I want.

### What did you see instead?
Because the index is in a narrow strip on the left, and because functions' returns are no longer in the index; it's very hard to see an overview of the package, because:
* You can't see the returns of a function in the index. Often seeing a compact list of functions in the package or methods in a type with their arguments and returns is all you need to lead you in the direction of the function that does what you want.
* In the main view there is little visual cue where one type's methods stop and another type begins. In the index this is very clear due to the indent. Scanning the main page to see functions' returns is visual overload to me, I just want to do it in the index.

In summary, please put the index back near the top like it is on godoc.org.

**Comment by brackendawson** [comment]
This is the index I'm talking about. Usually just looking at this format of index is almost all I need to use a godoc. It's just a matter of seeing a function that takes what you have and returns what you want then clicking on it to check its docstring:
![image](https://user-images.githubusercontent.com/1909753/92597668-a62e5e80-f29f-11ea-9b9f-44aaf7561490.png)

**Comment by myitcv** [comment]
Related to (if not a dup of) https://github.com/golang/go/issues/40612.

**Comment by brackendawson** [comment]
> Related to (if not a dup of) #40612.

I think my concern is entirely a subset of that issue, thanks.
